File: src/main/java/eu/qwsome/sql/ValueLiteral.java
package eu.qwsome.sql;

import eu.qwsome.sql.condition.ValueHolder;

/**
 * This class represents a literal value in sql.
 * <p>
 * For example: column = 1
 * where 1 is a literal.
 *
 * @author Lukáš Kvídera
 */
public class ValueLiteral implements ValueHolder {

  private final Object value;

  private ValueLiteral(final Object value) {
    this.value = value;
  }

  public static ValueLiteral value(final Object value) {
    return new ValueLiteral(value);
  }

  @Override
  public String getSql() {
    return "?";
  }

  @Override
  public Object getValue() {
    return this.value;
  }

  @Override
  public ValueHolder apply(final String functionName) {
    return new FunctionWrappedHolder(functionName, this);
  }
}

File: src/main/kotlin/eu/qwsome/sql/ValueLiteral.kt
/*
 * This class represents a literal value in sql.
 *
 * For example: column = 1
 * where 1 is a literal.
 *
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql

import eu.qwsome.sql.condition.ValueHolder
import eu.qwsome.sql.condition.withFunction

class ValueLiteral private constructor(private val value: Any?) : ValueHolder {

    override fun getSql(): String = "?"

    override fun getValue(): Any? = value

    override fun apply(functionName: String): ValueHolder = withFunction(functionName)

    companion object {
        @JvmStatic
        fun value(value: Any?) = ValueLiteral(value)
    }
}
